xchange-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioStreamingMarketDataService.java

```diff
@@ -1,18 +1,74 @@
 package info.bitrich.xchangestream.cexio;
 
+import info.bitrich.xchangestream.cexio.dto.CexioWebSocketOrderBookSubscribeResponse;
 import info.bitrich.xchangestream.core.StreamingMarketDataService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
 import io.reactivex.Observable;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Date;
+
 import org.knowm.xchange.currency.CurrencyPair;
 import org.knowm.xchange.dto.marketdata.OrderBook;
 import org.knowm.xchange.dto.marketdata.Ticker;
 import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.dto.trade.LimitOrder;
 import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
 
-public class CexioStreamingMarketDataService implements StreamingMarketDataService {
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
-    @Override
+public class CexioStreamingMarketDataService implements StreamingMarketDataService {
+	
+	private final CexioStreamingRawService streamingOrderDataService;
+	
+    public CexioStreamingMarketDataService(CexioStreamingRawService streamingOrderDataService) {
+		this.streamingOrderDataService = streamingOrderDataService;
+	}
+    
+    static class OrderBookUpdateConsumer implements io.reactivex.functions.Function<CexioWebSocketOrderBookSubscribeResponse, OrderBook> {
+    	BigInteger prevID = null;
+    	OrderBook orderBookSoFar = new OrderBook(new Date(), new ArrayList<LimitOrder>(), new ArrayList<LimitOrder>());
+    	final CexioStreamingRawService streamingOrderDataService;
+    	
+    	public OrderBookUpdateConsumer(CexioStreamingRawService streamingOrderDataService) {
+    		this.streamingOrderDataService = streamingOrderDataService;
+    	}
+    	
+		@Override
+		public OrderBook apply(CexioWebSocketOrderBookSubscribeResponse t) throws Exception {
+			OrderBook retVal;
+			if (prevID == null || prevID.add(BigInteger.ONE).compareTo(t.id) == 0) {
+        		prevID = t.id;
+        		orderBookSoFar = CexioAdapters.adaptOrderBookIncremental(orderBookSoFar, t);
+        		retVal = orderBookSoFar;
+        	} else {
+        		streamingOrderDataService.resubscribeChannels();
+        		retVal = orderBookSoFar;
+        		orderBookSoFar = new OrderBook(new Date(), new ArrayList<LimitOrder>(), new ArrayList<LimitOrder>());
+        		prevID = null;
+        	}
+			return retVal;
+		}
+    }
+    
+	@Override
     public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
-        throw new NotYetImplementedForExchangeException();
+		
+		streamingOrderDataService.waitForAuthIfNeeded();
```

Comment:
Why are you authenticating?  Most exchanges don't need it for public data like order books.

Might be worth having a read of #274 to make sure this is heading in the same direction (it looks like it is, at a glance)

Reply:
For some reason Cex.io requires users to authenticate before gaining access to the order book. This change seems to follow the logic outlined in #274 so far as elevating this stream to an authenticated one, but the mechanism isn't very clean. Since all Cex.IO APIs require authentication, I'll use the connect completable to signal that the authentication has completed.